phoenix-core/src/it/java/org/apache/phoenix/end2end/json/JsonFunctionsIT.java

```diff
@@ -633,4 +637,64 @@ public void testJsonQuery() throws Exception {
             compareJson(rs.getString(5), JsonDatatypes, "$.datatypes.mixedArray");
         }
     }
+
+    @Test
+    public void testJsonExpressionIndex() {
+        Properties props = PropertiesUtil.deepCopy(TEST_PROPERTIES);
+        String tableName = generateUniqueName();
+        String indexName = "IDX_" + generateUniqueName();
+        try (Connection conn = DriverManager.getConnection(getUrl(), props)) {
+            conn.setAutoCommit(true);
+            String
+                    ddl =
+                    "create table if not exists " + tableName +
+                            " (pk integer primary key, col integer, jsoncol.jsoncol json)";
+            conn.createStatement().execute(ddl);
+            conn.createStatement().execute(
+                    "UPSERT INTO " + tableName + " (pk, col, jsoncol) VALUES (1,2, '" + JsonDoc1 + "')");
+            conn.createStatement().execute(
+                    "CREATE INDEX " + indexName + " ON " + tableName
```

Comment:
Another thing to test is drop the json column from the base table. That should drop the index if any scalar field of json column is part of index pk. See `DropColumnIT#testDroppingIndexedColDropsIndex`